src/charm.py

```diff
@@ -770,6 +867,8 @@ def _on_update_status(self, event: UpdateStatusEvent):
         if not self.db_initialised:
             return
 
+        self.upgrade._reconcile_upgrade(event)
```

Comment:
why is this needed here? mind adding a comment?

Reply:
See above answer :)